compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1235,8 +1254,14 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
       super.ensureAccessible(tpe, superAccess, pos)
     }
 
+    override def typed(tree: untpd.Tree, pt: Type = WildcardType)(using Context): Tree =
```

Comment:
In fact, it might be better to do this test only in typedApply and typedTypeApply. Concentrating everything in `typed` is more elegant but might have performance implications. I am also nervous about making calls to `typed` more megamorphic.

Reply:
I moved the logic back to `typedApply`, `typedTypeApply`, `typedIdent` and `typedSelect`. I kept the rest of the refactoring which avoids the duplication of the code that is now in `needsInlining`.